### DISCUSSION TITLE
[QUESTION]How to use signalingClientFetchSync on battery devices (can't receive SDP Offer after wake up)

### DISCUSSION THREAD

**Discussion by 1045672215** [awslabs/amazon-kinesis-video-streams-webrtc-sdk-c#1936] [Q&A]
We use the webRTC on battery devices with sleep and wake up functions and want to achieve that devices are able to watch living streaming immediately after waking up.

** Any design considerations/constraints **
Before v1.7.0, we call freeSignalingClient function before sleep and call createSignalingClientSync function after wake up.It works.But when we update the sdk after v1.7.0, we only call signalingClientFetchSync function before sleep, we find devices can't receive SDP Offer after wake up.
(Devices has been connected to the signaling server and can receive ICE candidates.)We try to call signalingClientFetchSync function after wake up,but it didn't solve the problem.
So we want to know how to use signalingClientFetchSync on battery devices to solve the problem.

**Comment by disa6302** [comment]
@1045672215 ,

Can you confirm if you invoke `signalingClientConnectSync` after `signalingClientFetchSync`? The signaling connection duration has a hard 1 hour limit. Wondering if you are hitting the limit during wake up.

**Reply by 1045672215** [reply]
We do invoke signalingClientConnectSync after signalingClientFetchSync.And we aren't hitting the limit during wake up.Here my code about this part.
![image](https://github.com/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/assets/42050600/2278c92d-d393-4da3-b277-ee7fbd4d9994)

**Reply by 1045672215** [reply]
[log.txt](https://github.com/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/files/14415191/log.txt)
This is the log of test.After signalingClientFetchSync and signalingClientConnectSync,the device has connected signaling server,but only receives ICE candidates and does not receive SDP Offer.

**Comment by disa6302** [comment]
@1045672215 , 

What is your viewer look like? I am wondering if the viewer sent the offer but the C SDK was not ready to receive it and it got lost.

**Reply by 1045672215** [reply]
Yes.We see the log of viewer,and find viewer has already sent SDP Offer.But Master doesn't receive it.We are wondering to know if Signaling server doesn't forward the SDP Offer, or it does, but C SDK is not ready to receive it.If it is the latter, what is the reason? After all, from the logs, C SDK has already connected to the Signaling server.How should we solve this problem？

**Reply by 1045672215** [reply]
Moreover,we have tried various APIs to address the issue of not receiving SDP Offer after wake up.We speculate that it was due to the unsuccessful disconnection and release of the previous connection.We find that signalingClientDisconnectSync function was called before sleep, and signalingClientConnectSync function was called after waking up, or freeSignalingClient function was called before sleep ,and createSignalingClientSync+signalingClientFetchSync+signalingClientConnectSync were called after waking up camera can receive SDP Offer sometimes.But most of the time it's not received.Is this issue related to the use of interface functions?
